_data-prepper/core-apis.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: default
+title: Core APIs
+nav_order: 2
+---
+
+# Core APIs
+
+All Data Prepper instances expose a server with some control APIs. By default, this server runs on port 4900. Some plugins, especially source plugins may expose other servers. These will be on different ports, and their configurations are independent of the core API. For example, to shut down Data Prepper, you can run the following curl request:
```

Comment:
1. Just for readability, suggestion:
"Some plugins, especially source plugins, may expose other servers that run on different ports. Configurations for these plugins are independent of the core API."

2. Is shutting down Data Prepper with a cURL command an example of the statement about "independent configurations" that precedes it? Is "configurations" the right word here? Or is it the way you format calls to the server that are unique to each plugin (including Data Prepper control APIs)? Would something like this be more accurate?
"Some plugins, especially source plugins, may expose other servers that run on different ports. Therefore, the API requests for these plugins and Data Prepper will vary according to the port used for communication. For example, to shut down Data Prepper, you can run the following curl request:"

Reply:
@oeyh Can you please clarify on the second comment here?

Reply:
> Is shutting down Data Prepper with a cURL command an example of the statement about "independent configurations" that precedes it?

It's not. I believe it's just an example of the statement "Data Prepper expose a server with some control APIs. By default, this server runs on port 4900."

> Is "configurations" the right word here? Or is it the way you format calls to the server that are unique to each plugin (including Data Prepper control APIs)?

The word "configurations" sounds right to me. We use "configurations" to refer to "settings" for the plugins, which may include a port number. Plugins and core API have separate configurations, and specify different ports in their own configurations.